### PR TITLE
Support colorise on Windows console

### PR DESCRIPTION
**Pull request by koron**
How about support colorise (escape sequence) on Windows console with [mattn/go-colorable](https://github.com/mattn/go-colorable)?

### CODE REVIEW

**Comment by yongpingchen** [comment]
+1

**Comment by robdimsdale** [comment]
@koron in principle this looks good. I have two questions. One, does this change anything for non-windows users? The library looks like it handles windows-specific output, so I would expect it to have no impact on other operating systems.

The second thing is we have now introduced a new external dependency. This means we will no longer be able to `go get github.com/onsi/ginkgo/ginkgo` and expect the binary to be present. We can, however, use:

```
go get github.com/onsi/ginkgo/...
go get github.com/onsi/ginkgo/ginkgo
```

to install the binary.

I opened an issue at https://github.com/onsi/ginkgo/issues/240 to capture this - I think we need to answer some of these questions before we can merge this PR.

Thanks,
Rob

File: reporters/stenographer/stenographer.go [hunk 1]
```diff
@@ -8,9 +8,11 @@ package stenographer
 
 import (
 	"fmt"
+	"io"
 	"runtime"
 	"strings"
 
+	"github.com/mattn/go-colorable"
```
Comment (jayunit100) [inline reporters/stenographer/stenographer.go]: hmmm.  i see why this is needed, but i think  this just a single file ? if so i wonder if ginkgo should really just carry it, ... b/c as of now there is no support or need for GoDep.

**Comment by onsi** [comment]
I'd like to avoid using Godep and I'd like to avoid third party dependencies, please.  There's not a lot of complexity in go-colorable and I don't think it's going to be a fast-moving project/burden to maintain.  I'd like us to copy the repo in directly in much the same way that [Gomega does with goraph](https://github.com/onsi/gomega/tree/master/matchers/support/goraph)

I propose a `reporters/stenographer/support/go-colorable` package that pulls in go-colorable with the license file and a README.md with appropriate attribution and al ink back to the original GitHub page.

**Comment by onsi** [comment]
Also.  **THANK YOU FOR THIS** - I kid you not Ginkgo started because I missed the little-green-dots that [Cedar](https://github.com/pivotal/cedar), [Jasmine](https://github.com/jasmine/jasmine) and [rspec] (https://github.com/rspec/rspec) gave me.  Having the green dots not work on windows was always sad... happy to see it get fixed with this PR!

**Comment by robdimsdale** [comment]
@onsi this pattern of copying dependencies under a custom path works but it's not as discoverable as locating them in the `vendor` directory at the root level and delegating to `GO15VENDOREXPERIMENT`. This allows us to use the original import path rather than some other, custom path.

I don't feel strongly about where dependencies live, but I agree that libraries (and, to a lesser extent, binaries) should compile out of the box with `go get`.

**Comment by koron** [comment]
@robdimsdale 

> does this change anything for non-windows users?

Yes.  As you seen, go-colorable does noting for other environments.
(cf <https://github.com/mattn/go-colorable/blob/master/colorable_others.go#L18-L20>)

> The second thing is we have now introduced a new external dependency.

It is a point that I was concerned.

Using golang's native vendoring is better, I think.
If @onsi agree it, I'll update this PR to use vendoring.
(Just copy to vendor/github.com/mattn/go-colorable :wink:)

**Comment by koron** [comment]
Ah It is off topic but

    go get github.com/onsi/ginkgo

This will download all packages which ginko depends on.

**Comment by robdimsdale** [comment]
@onsi in the interests of getting this PR merged, I'm happy to vendor in whichever way you prefer.

@koran can you submit the PR with @onsi suggestion above so we can look at it and merge it in? I know we all want to see this PR merged.

Thanks,
Rob

**Comment by koron** [comment]
@onsi and @robdimsdale.  I'm asking to author (@mattn) to include him packages to this project with some changes.
I'll fix PR as onsi said, after him response.

**Comment by mattn** [comment]
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::smile::smile::smile::smile::smile::+1:
:+1::+1::+1::+1::+1::+1::+1::+1:
:+1::+1::smile::smile::smile::smile::+1::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::smile::smile::smile::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::+1::+1::+1::smile::smile::+1:
:+1::+1::smile::smile::smile::+1::smile::+1:
:+1::+1::+1::+1::+1::+1::+1::+1:
:+1::smile::smile::smile::smile::smile::smile::smile:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::smile::+1::+1::+1:
:+1::+1::+1::+1::+1::+1::+1::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::smile::+1::+1::smile::smile::+1:
:+1::smile::smile::+1::+1::smile::smile::+1:
:+1::smile::+1::smile::smile::+1::smile::+1:
:+1::smile::+1::smile::smile::+1::smile::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:
:+1::smile::+1::+1::+1::+1::smile::+1:

**Comment by koron** [comment]
@onsi 

> I propose a reporters/stenographer/support/go-colorable package that pulls in go-colorable with the license file and a README.md with appropriate attribution and al ink back to the original GitHub page.

I did it with d2481fa.  It includes go-isatty which is required by go-colorable and also created by @mattn.
If you prefer to rename `support` dir to `supports`, I'll do it.

**Comment by aubm** [comment]
Any updates on this? 😄

**Comment by onsi** [comment]
welp.  sorry this fell through the craks!  looks good to me.

**Comment by onsi** [comment]
merged now!

**Comment by koron** [comment]
thanks 🙇

**Pull request merged**